watch_service.go

```diff
@@ -67,8 +72,11 @@ func (ws *watchService) watchFile(ctx context.Context, projectName, repoName, la
 func (ws *watchService) watchRepo(ctx context.Context,
 	projectName, repoName, lastKnownRevision, pathPattern string, timeout time.Duration) <-chan *WatchResult {
 	watchResult := make(chan *WatchResult)
-	if len(pathPattern) != 0 && !strings.HasPrefix(pathPattern, "/") {
-		// Normalize the pathPattern when it does not start with "/" so that the pathPattern fits into the url.
+
+	// Normalize pathPattern
+	if len(pathPattern) == 0 || pathPattern == "**" {
```

Comment:
Thanks for adding a case that the previous code didn't cover. 👍 

But I think we should split it:
```
if len(pathPattern) == 0 {
    pathPattern = "/**"
} else if strings.HasPrefix(pathPattern, "**") {
    pathPattern = "/" + pathPattern
} else if !strings.HasPrefix(pathPattern, "/") {
    pathPattern = "/**/" + pathPattern
}
```

Reply:
Thank you for quick review.
Because normalizing malform `pathPattern` from user has a small number of fixed logic branching. 
And I think it's quite clear while reading code with only one if-else rather than separating. WDYT @minwoox

Reply:
Because you handle the case which the `pathPattern` is just `**`, I think we should handle the cases which the `pathPattern` start with `**` such as `**/foo.json`.
Sorry about my poor explanation in the above comment. :)

Reply:
And rather than: `pathPattern = "/" + pathPattern`, we could use string literal `/**` to prevent memory allocation.

Reply:
Please pardon my comment above. I wonder if `/**/**/foo.json` is still correct 🤔 ?

Reply:
Well, I think `/**/foo.json` would be better? (I'm not sure those two `pathPattern`s works in the same way or not :-) )

Reply:
🤔 sound reasonable. I have just committed to handle `pathPattern` start with `**` as your suggestion. PTAL @minwoox san 🙇